components/dao-agent/src/context.rs

```diff
@@ -22,107 +21,138 @@ sol! {
-            // Only support http/https and ipfs URIs
-            Err(format!("Unsupported URI scheme: {}", clean_uri))
-        }
+    pub fn load_from_uri(uri: &str) -> Result<Self, String> {
+        block_on(async move {
+            // Strip any quotation marks from the URI
+            let clean_uri = uri.trim_matches('"');
+
+            println!("Loading config from URI: {}", clean_uri);
+
+            // Check URI scheme
+            if let Some(uri_with_scheme) = clean_uri.strip_prefix("ipfs://") {
+                // IPFS URI scheme detected
+                Self::load_from_ipfs(uri_with_scheme)
+            } else if clean_uri.starts_with("http://") || clean_uri.starts_with("https://") {
+                // HTTP URI scheme detected
+                Self::fetch_from_uri(clean_uri)
+            } else {
+                // Only support http/https and ipfs URIs
+                Err(format!("Unsupported URI scheme: {}", clean_uri))
+            }
+        })
     }
 
     /// Load configuration from IPFS
-    async fn load_from_ipfs(cid: &str) -> Result<Self, String> {
-        let gateway_url = std::env::var("WAVS_ENV_IPFS_GATEWAY_URL").unwrap_or_else(|_| {
-            println!("WAVS_ENV_IPFS_GATEWAY_URL not set, using default");
-            "https://gateway.lighthouse.storage/ipfs".to_string()
-        });
-
-        // Strip any quotation marks from the gateway URL
-        let clean_gateway_url = gateway_url.trim_matches('"');
-
-        // Construct HTTP URL, avoiding duplicate /ipfs in the path
-        let http_url = if clean_gateway_url.ends_with("/ipfs") {
-            format!("{}/{}", clean_gateway_url, cid)
-        } else if clean_gateway_url.ends_with("/ipfs/") {
-            format!("{}{}", clean_gateway_url, cid)
-        } else if clean_gateway_url.ends_with("/") {
-            format!("{}ipfs/{}", clean_gateway_url, cid)
-        } else {
-            format!("{}/ipfs/{}", clean_gateway_url, cid)
-        };
+    fn load_from_ipfs(cid: &str) -> Result<Self, String> {
+        block_on(async move {
+            let gateway_url = std::env::var("WAVS_ENV_IPFS_GATEWAY_URL").unwrap_or_else(|_| {
+                println!("WAVS_ENV_IPFS_GATEWAY_URL not set, using default");
+                "https://gateway.lighthouse.storage/ipfs".to_string()
+            });
+
+            // Strip any quotation marks from the gateway URL
+            let clean_gateway_url = gateway_url.trim_matches('"');
+
+            // Construct HTTP URL, avoiding duplicate /ipfs in the path
+            let http_url = if clean_gateway_url.ends_with("/ipfs") {
+                format!("{}/{}", clean_gateway_url, cid)
+            } else if clean_gateway_url.ends_with("/ipfs/") {
+                format!("{}{}", clean_gateway_url, cid)
+            } else if clean_gateway_url.ends_with("/") {
+                format!("{}ipfs/{}", clean_gateway_url, cid)
+            } else {
+                format!("{}/ipfs/{}", clean_gateway_url, cid)
+            };
```

Comment:
Maybe simpler? (not validated this logic, just cleanup for easier understanding)

```suggestion
let http_url = {
    let base = clean_gateway_url.trim_end_matches('/');
    let ipfs_part = if base.ends_with("/ipfs") {
        "/"
    } else {
        "/ipfs/"
    };
    format!("{}{}{}", base, ipfs_part, cid)
};
```